src/Control/Concurrent/Supervisor.hs

```diff
@@ -27,4 +27,6 @@ type Child = Types.Child_ TQueue
 --------------------------------------------------------------------------------
 -- NOTE: The `maxBound` value will be ignore by the underlying implementation.
 newSupervisor :: RestartStrategy -> IO Supervisor
-newSupervisor str = Types.newSupervisor str maxBound
+newSupervisor str = Types.newSupervisor str 9223372036854775807
```

Comment:
How come you had to change this line?

Reply:
aha, I see! Thanks for the clarification :)

By all means, feel free to fire a second PR with any improvement you feel like adding 😉